domain/src/main/java/org/oppia/domain/repository/SkillRepository.kt

```diff
@@ -0,0 +1,209 @@
+import org.oppia.app.model.ConceptCardView
+import org.oppia.app.model.LessonThumbnail
+import org.oppia.app.model.LessonThumbnailGraphic
+import org.oppia.app.model.SkillSummaryDatabase
+import org.oppia.app.model.SkillSummaryDomain
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.domain.topic.FRACTIONS_SKILL_ID_0
+import org.oppia.domain.topic.FRACTIONS_SKILL_ID_1
+import org.oppia.domain.topic.FRACTIONS_SKILL_ID_2
+import org.oppia.domain.topic.FRACTIONS_TOPIC_ID
+import org.oppia.domain.topic.RATIOS_SKILL_ID_0
+import org.oppia.domain.topic.RATIOS_TOPIC_ID
+import org.oppia.domain.topic.TOPIC_FILE_ASSOCIATIONS
+import org.oppia.domain.util.JsonAssetRetriever
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProvider
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+
+private const val ADD_SKILL_SUMMARY_LIST_TRANSFORMED_PROVIDER_ID = "add_skill_list_transformed_id"
+private const val GET_SKILL_SUMMARY_TRANSFORMED_PROVIDER_ID = "get_skill_summary_transformed_id"
+
+/** Controller for retrieving all aspects of a topic. */
+@Singleton
+class SkillRepository @Inject constructor(
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val dataProviders: DataProviders,
+  private val jsonAssetRetriever: JsonAssetRetriever,
+  private val logger: Logger
+) {
+
+  /** Indicates that the given skill summary already exists. */
+  class SkillSummaryAlreadyExistsException(msg: String) : Exception(msg)
+
+  /** Indicates that the given skill summary is not found. */
+  class SkillSummaryNotFoundException(msg: String) : Exception(msg)
+
+  /**
+   * These statuses correspond to the exceptions above such that if the deferred contains
+   * SKILL_SUMMARY_ALREADY_FOUND, the [SkillSummaryAlreadyExistsException] will be passed to a failed AsyncResult.
+   *
+   * SUCCESS corresponds to a successful AsyncResult.
+   */
+  private enum class SkillSummaryActionStatus {
+    SUCCESS,
+    SKILL_SUMMARY_ALREADY_EXISTS,
+    SKILL_SUMMARY_NOT_FOUND
+  }
+
+  private val skillSummaryDataStore =
+    cacheStoreFactory.create(
+      "skill_summary_database",
+      SkillSummaryDatabase.getDefaultInstance()
+    )
+
+  // TODO(#272): Remove init block when storeDataAsync is fixed
+  init {
+    skillSummaryDataStore.primeCacheAsync().invokeOnCompletion {
+      it?.let {
+        logger.e(
+          "DOMAIN",
+          "Failed to prime cache skillSummaryDataStore ahead of LiveData conversion for SkillRepository.",
+          it
+        )
+      }
+    }
+  }
+
+  /**
+   * This method stores all skills to proto database by reading these skills from corresponding json files
+   * and returns a successful [LiveData]
+   */
+  fun storeAllSkillSummariesToProtoDatabase(): LiveData<AsyncResult<Any?>> {
+    return MutableLiveData(
+      try {
+        TOPIC_FILE_ASSOCIATIONS.keys.forEach { topicId ->
+          addSkillSummaryList(retrieveSkillSummaryDomainListFromJson(topicId))
```

Comment:
1. By first if you mean that we should combine all skills across all topics and send it to `addSkillSummaryList` (now `insertSkillSummaryList`) for inserting them to proto-database, then in that case this is correct because in future when API call will be implemented a list of skills will be coming for a particular topic.

2. Updated code to make `retrieveSkillSummaryDomainListFromJson` as dataprovider. PTAL

Reply:
What issue are you observing? I was thinking that you would convert each insert call would be its own data provider, then you combine them all together. You could alternatively move the for loop into a single data provider suspend callback to push all of that to a background thread. Would either of these work?